Feat: add linting rule for before/beforeEach just like the async test rule (#151)
cypress-io/eslint-plugin-cypress: docs/rules/no-async-before.md

# Prevent using async/await in Cypress test cases (no-async-tests)

> Cypress before that return a promise will cause tests to prematurely start, can cause unexpected behavior. An `async` function returns a promise under the hood, so a before using an `async` function might also error.

Reviewer: are we sure this is always the case? This might be true for certain commands but I don't think generally this applies.

Author: I do not know if this is always true, I guess the cypress team knows this better than me. I noticed that the test method was already being executed while the before was still in progress

Reviewer: interesting. I don't think I have run into this but I will take your word for it. Knowing how mocha hooks process and how we integrate with them, I can absolutely see this being true

> Cypress before that return a promise will cause tests to prematurely start, can cause unexpected behavior. An `async` function returns a promise under the hood, so a before using an `async` function might also error.

Reviewer: I wonder if there is a better way to phrase this? Can we find any concrete examples where before/beforeEach awaiting cypress commands cause an error similar to test bodies?
```suggestion
Cypress commands that return a promise may cause side effects in before/beforeEach hooks, possibly causing unexpected behavior. 
```

## Rule Details

> This rule disallows using `async` before and beforeEach functions.

Reviewer: ```suggestion
This rule disallows using `async` `before` and `beforeEach` functions.
```

Reviewer 2: ```suggestion
This rule disallows using `async` in `before` and `beforeEach` hook functions.
```

Sorry, @AtofStryker , just a slight update to the suggestion :)
